synapse/rest/synapse/client/__init__.py

```diff
@@ -42,6 +42,18 @@ def build_synapse_client_resource_tree(hs: "HomeServer") -> Mapping[str, Resourc
         "/_synapse/client/sso_register": SsoRegisterResource(hs),
     }
 
+    # provider-specific SSO bits. Only load these if they are enabled, since they
+    # rely on optional dependencies.
+    if hs.config.oidc_enabled:
+        from synapse.rest.oidc import OIDCResource
+
+        resources["/_synapse/oidc"] = OIDCResource(hs)
+
+    if hs.config.saml2_enabled:
+        from synapse.rest.saml2 import SAML2Resource
+
+        resources["/_matrix/saml2"] = SAML2Resource(hs)
```

Comment:
Does it make sense to note that this is under `_matrix` for backwards compat?

Reply:
sorry, note where? as a comment here?

Reply:
As a comment, yes. (My thinking being that this is in the `synapse.rest.synapse` package someone might be confused in the future at why we're registering a `_matrix` URL here. 🤷 ) Not a big deal either way though.

Reply:
ok, I added a comment.